### PR TITLE
Feature/slepukhin/wizard order

### PR DESCRIPTION
**Pull request by vitslepukhin**

### CODE REVIEW

File: src/app/core/services/location/professional-location.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { DistrictApiService } from '@app/core/services/location/district-api.service';
+import { RegionApiService } from '@app/core/services/location/region-api.service';
+import { SubregionApiService } from '@app/core/services/location/subregion-api.service';
+import { City } from '@app/profile/models/city';
+import { Country } from '@app/profile/models/country';
+import { forkJoin, Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import {ProfessionalLocationInline} from '../../../api/models';
+
+export type FullLocation = {
```
Comment (enkryptor) [inline src/app/core/services/location/professional-location.service.ts:16]: separate file?

File: src/app/order/order-steps.ts [hunk 2]
```diff
@@ -1,2 +1,101 @@
...
+    title: string;
+}
+
+export interface StepsModel {
+    byId: { [id: string]: StepModel };
+    ids: string[];
+}
+
+export type StepState<T extends any> = {
+    isComplete: boolean;
```
Comment (enkryptor) [inline src/app/order/order-steps.ts:26]: `isValid` ?

**Review by webmalc — APPROVED** [review]

File: src/app/order/components/step-container/step-container.component.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    public submit(): void {
+        this.wizardState.doSubmit();
+    }
+
+    private subscribeInputCurrentState(): void {
+        this.wizardState
+            .getCurrentStepState()
+            .pipe(take(1), takeUntil(this.ngDestroy$))
```
Comment (enkryptor) [inline src/app/order/components/step-container/step-container.component.ts:80]: let's emit state change on "next" button pressed

File: src/app/order/order-steps.ts [hunk 4]
```diff
@@ -1,2 +1,101 @@
...
+            component: ClientDetailsStepComponent,
+            title: 'order.step.client-details'
+        },
+        [OrderIds.Summary]: {
+            id: OrderIds.Summary,
+            component: SummaryStepComponent,
+            title: 'order.step.summary'
+        }
+    },
+    ids: [
```
Comment (enkryptor) [inline src/app/order/order-steps.ts:83]: `order`?

**Review by enkryptor — CHANGES_REQUESTED** [review]
update step state only when "Next" button is pressed

File: src/app/core/services/location/professional-location.service.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import { CitiesApiService } from '@app/core/services/location/cities-api.service';
+import { CountriesApiService } from '@app/core/services/location/countries-api.service';
+import { DistrictApiService } from '@app/core/services/location/district-api.service';
+import { RegionApiService } from '@app/core/services/location/region-api.service';
+import { SubregionApiService } from '@app/core/services/location/subregion-api.service';
+import { City } from '@app/profile/models/city';
+import { Country } from '@app/profile/models/country';
+import { forkJoin, Observable, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import {ProfessionalLocationInline} from '../../../api/models';
```
Comment (enkryptor) [inline src/app/core/services/location/professional-location.service.ts:14]: please use absolute paths

File: src/app/order/abstract/step.ts [hunk 6]
```diff
@@ -1,19 +1,41 @@
...
+} from '@angular/core';
+import { Subject } from 'rxjs';
+import { StepContext, StepState } from '../order-steps';
 
-export abstract class StepComponent {
-    protected constructor(protected readonly wizardState: OrderWizardStateService) {
+@Component({
+    template: ''
+})
+export abstract class StepComponent<T extends any> implements OnDestroy {
```
Comment (enkryptor) [inline src/app/order/abstract/step.ts:13]: just `T` is enough

**Review by ennaque — APPROVED** [review]

**Review by enkryptor — APPROVED** [review]

**Pull request merged**